projects/go-lib/src/lib/components/go-portal/go-portal.service.ts

```diff
@@ -0,0 +1,31 @@
+import {Injectable, TemplateRef, ViewContainerRef} from '@angular/core';
+
+@Injectable({providedIn: 'root'})
+export class GoPortalService {
+  private targets: Map<string, ViewContainerRef>;
+
+  constructor() {
+    this.targets = new Map<string, ViewContainerRef>();
+  }
+
+  addTarget(targetName: string, viewContainer: ViewContainerRef): void {
+    this.targets.set(targetName, viewContainer);
+  }
+
+  attachToTarget(targetName: string, template: TemplateRef<any>): void {
+    if (this.targets.has(targetName)) {
+      this.getTarget(targetName).createEmbeddedView(template);
+    }
+  }
```

Comment:
Seeing as how we already do this `if (this.targets.has(targetName)) {` inside of the `getTarget` method below, we can simplify this a bit and remove an extra lookup on the map:

```
this.getTarget(targetName)?.createEmbeddedView(template);
```

Reply:
Optional chaining is only available in TypeScript 3.7, but we are compiling with 3.5, so that ain't gonna work.

Reply:
Nuts, but I still think we can make it slightly more performant by not doing another lookup on the map.

```
const target: ViewContainer = this.getTarget(targetName);

if (target) {
  target.createEmbeddedView(template);
}
```

Thoughts?

Reply:
Sounds like a micro optimization to me, but sure I'll do that.